src/ompl/geometric/planners/bitstar/datastructures/src/IntegratedQueue.cpp

```diff
@@ -0,0 +1,1693 @@
+                this->rmOutgoingLookup(oldEdgeIter);
+            }
+            //No else
+
+            //Finally erase from the queue:
+            edgeQueue_.erase(oldEdgeIter);
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmIncomingLookup(const edge_queue_iter_t& mmapIterToRm)
+        {
+            if (incomingLookupTables_ == true)
+            {
+                this->rmEdgeLookupHelper(incomingEdges_, mmapIterToRm->second.second->getId(), mmapIterToRm);
+            }
+            //No else
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmOutgoingLookup(const edge_queue_iter_t& mmapIterToRm)
+        {
+            if (outgoingLookupTables_ == true)
+            {
+                this->rmEdgeLookupHelper(outgoingEdges_, mmapIterToRm->second.first->getId(), mmapIterToRm);
+            }
+            //No else
+        }
+
+
+
+        void BITstar::IntegratedQueue::rmEdgeLookupHelper(vid_edge_queue_iter_umap_t& lookup, const BITstar::vid_t& idx, const edge_queue_iter_t& mmapIterToRm)
+        {
+            //Variable:
+            //An iterator to the vertex,list pair in the lookup
+            vid_edge_queue_iter_umap_t::iterator iterToVertexListPair;
+
+            //Get the list in the lookup for the given index:
+            iterToVertexListPair = lookup.find(idx);
+
+            //Make sure it was actually found before derefencing it:
+            if (iterToVertexListPair != lookup.end())
+            {
+                //Variable:
+                //Whether I've found the mmapIterToRm in my list:
+                bool found;
+                //The iterator to the mmapIterToRm in my list:
+                edge_queue_iter_list_t::iterator iterToList;
+
+                //Start at the front:
+                iterToList = iterToVertexListPair->second.begin();
+
+                //Iterate through the list and find mmapIterToRm
+                found = false;
+                while(found == false && iterToList != iterToVertexListPair->second.end())
+                {
+                    //Compare the value in the list to the target:
+                    if (*iterToList == mmapIterToRm)
+                    {
+                        //Mark as found:
+                        found = true;
+                    }
+                    else
+                    {
+                        //Increment the iterator:
+                        ++iterToList;
+                    }
+                }
+
+                if (found == true)
+                {
+                    iterToVertexListPair->second.erase(iterToList);
+                }
+                else
+                {
+                    throw ompl::Exception("Edge iterator not found under given index in lookup hash.");
+                }
+            }
+            else
+            {
+                throw ompl::Exception("Indexing vertex not found in lookup hash.");
+            }
+        }
+
+
+
+
+
+
+
+
+
+
```

Comment:
Right.
So I do try to be consistent with white space, so when there are large gaps like this, it isn't out of laziness/sloppiness. My logic is as follows:
 - 1 line of white space inside functions. I find this helps break up the code and make things more readable.
 - 3 lines of white space between functions. This is to help identify where functions start and stop as there are single white space lines inside functions. I find this especially helpful when scrolling around in a file.
 - A bunch of white space between "groups" of functions. So for example between public and private functions (what this is), or before a bunch of really boring sets/gets (which I always try to hide at the bottom of a file).